test/quantization/dbr/test_quantize_dbr.py

```diff
@@ -1122,6 +1123,46 @@ def test_qconfig_dict_module_name(self):
         self.assertTrue(isinstance(mq[2][0], nn.Conv2d))
         self.assertTrue(isinstance(mq[2][1], nnq.Conv2d))
 
+    def test_serialization(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.conv = torch.nn.Conv2d(1, 1, 1)
+                self.linear = torch.nn.Linear(1, 1)
+
+            def forward(self, x):
+                x1 = self.conv(x)
+                x2 = self.linear(x1)
+                return x2
+
+        input_shape = (1, 1, 1, 1)
+        qconfig_dict = {'': torch.quantization.default_qconfig}
+        example_inputs = (torch.randn(*input_shape),)
+
+        m = M().eval()
+        m = _quantize_dbr.prepare(m, qconfig_dict, example_inputs)
+        m = _quantize_dbr.convert(m)
+        m_copy = copy.deepcopy(m)
+
+        # Ensure the two models have different weights
```

Comment:
One thing to note is that currently module swaps are handled by Eager mode quantization, and DBR reuses that code.  The "new" code that DBR adds is automatic insertion of quants and dequants, and quantization of functions and methods.  The current test mostly tests the modules.  We could add two things here to make sure we also test the DBR-only logic:
1. we can calibrate with sample data before converting, this will ensure that the scale and zero_point of observers throughout the model are based on calibration data instead of default
2. if we change things manually to make the 2nd model be different from the 1st model, we should also change the internal state of observers which will be swapped by quants.  Or, a faster option could be to create the 2nd model from scratch but do not calibrate it.

Something like this could ensure that DBR-only functionality is also being tested:

```
m = M().eval()
m = _quantize_dbr.prepare(m, qconfig_dict, example_inputs)
# calibrate, to populate statistics
m(torch.randn(1, 1, 1, 1))
m = _quantize_dbr.convert(m)

# create new model
m2 = M().eval()
m2 = _quantize_dbr.prepare(m2, qconfig_dict, example_inputs)
# do not calibrate, to ensure important statistics are populated without calibration and 
# the results are different at every node, including the quantize_per_tensor node
m2 = _quantize_dbr.convert(m2)

# now, m and m2 are different, load the state dict and verify equivalency
```

Reply:
That makes sense. I updated the code accordingly. However, the test now fails. It seems there's something that we're not capturing in the `state_dict`, since I verified that both `state_dict`s are identical after serialization. I am investigating this further.